doc/user/content/transform-data/patterns/top-k.md

```diff
@@ -100,3 +100,27 @@ FROM cities
 ORDER BY state, pop DESC;
 ```
 Note that the `ORDER BY` clause should start with the expressions that are in the `DISTINCT ON`.
+
+## Group size hints
+
+When using either the above `LATERAL` subquery pattern or `DISTINCT ON`, we recommend specifying a [group size hint](/transform-data/optimization/#query-hints) to improve memory usage. For example:
+
+```sql
+SELECT state, name FROM
+    (SELECT DISTINCT state FROM cities) grp,
+    LATERAL (
+        SELECT name FROM cities
+        WHERE state = grp.state
+        OPTIONS (LIMIT INPUT GROUP SIZE = 1000)
+        ORDER BY pop DESC LIMIT 3
+    );
+```
+
+or
+
+```sql
+SELECT DISTINCT ON(state) state, name
+FROM cities
+OPTIONS (DISTINCT ON INPUT GROUP SIZE = 1000)
+ORDER BY state, pop DESC;
+```
```

Comment:
I'd suggest adding a sentence at the end like "To learn more about how to set the values for the query hints above, see the [Optimization](/transform-data/optimization/#query-hints) page."